README.md

```diff
@@ -424,7 +426,48 @@ Assumes database resolution is done by adding prefixes. If database resolution i
      where h.date = '2016-05-13'
        and h.hour = 1
     ;
+
+## Database Name Mapping
+Waggle Dance allows configuration to map database names. This feature is intended to be useful when migrating from an existing database into different environments.
+A use case:
+We have legacy data lakes that have tables belonging to the same domain in differently named databases. E.g. a booking domain is called 'datawarehouse' or 'hotel_booking'. When federating to those datalakes it helps data discovery if those names are standardized to one name. Renaming a database is easy but migrating ETL might not be, so to make migration easier you can configure Waggle Dance to remap the names.
+
+Example Datalake X federating to 2 other "legacy" datalakes Y and Z.
+This is what we want the different databases to look like in the datalakes:
+
+X has:
+
+    y_booking
+    z_booking
+
+Y has:
+    
+    datawarehouse
+    another_db
+
+Z has:
     
+    hotel_booking
+
+To achieve this we can configure Waggle Dance in X to map the names:
+
+
+    federated-meta-stores:
+      - remote-meta-store-uris: thrift://10.0.0.1:9083
+        name: y
+        mapped-databases:
+        - datawarehouse
+        database-name-mapping:
+          datawarehouse: booking
+      - remote-meta-store-uris: thrift://10.0.0.2:9083
+        name: z
+        database-name-mapping:
+          hotel_booking: booking
+
+
+If an optional `mapped-databases` is used that filter is applied first and the renaming is applied after.
+NOTE: mapping names adds an extra layer of abstraction and we advise to use this as a temporary migration solution only. It becomes harder to debug where a virtual (remapped) table actually is coming from.
```

Comment:
I'd move this to the top of the section to highlight this isn't something we recommend as early as possible.

Reply:
that's a good point :)